style_guide.md

```diff
@@ -0,0 +1,217 @@
+   lib/
+      common_rules.libsonnet
+      placeholders.libsonnet
+   ccs_household.jsonnet
+   census_communal_establishment.jsonnet
+   census_household.jsonnet
+   census_individual.jsonnet
+northern-ireland/
+
+```
+- England/Wales and Northern Ireland
+
+    - Jsonnet source files are separated into two folders: `england-wales` and `northern-ireland` with a number of sub-folders, each for corresponding schemas (e.g. `ccs`, `communal-establishment`, `household`, `individual`). These are divided into blocks in `blocks` folder.
+
+- Shared blocks
+
+    - Blocks from `individual` folder are used in both household and individual schema.
+
+- Top level Jsonnet files
+
+    - For each schema there are lists of block and placeholder imports.
+    - Schema structure is divided into sections, named after given schema and ending with `.jsonnet` suffix.
+
+- Lib folders
+
+    - Rules and placeholders are located in `lib` folder in corresponding schemas along with `blocks` folder.
+
+- Rules and placeholders
+
+    - These are separated into: `common_rules`, `placeholders` and `rules`. These files should end with `.libsonnet` suffix.
+
+- Top level and extended
+
+    - `common_rules` and `placeholders` are located in top-level "lib" folder.
+    - `rules` files for each schema are located in schema's sub-folder and extend some of the top-level `common_rules`.
+
+## Global variables
+
+- Top level arguments
+
+    - For each schema created it is possible to pass variables, e.g.: `region_code` and `census_month_year_date`, with different outputs being generated depending on their value.
+    - These are defined in eq-questionnaire-runner, are known at questionnaire launch and won't change during the Census.
+
+- External variables
+
+    - External variables are being used. Example of that is `census_month_year_date`.
+
+- Region Code
+
+    - Depending on `region_code` variable's value, different structure of sections is generated (e.g. certain questions omitted).
+
+## Blocks
+
+### Structure
+
+#### Block json
+- There are two distinct styles we use with block json:
+  - Simple (no variants):
+    ```
+    // Imports
+    
+    {
+      // Block JSON
+    }
+    ```
+  - Variants:
+    ```
+    // Imports
+    
+    // Variant helper methods
+    local questionTitle(variantParam) = (
+      // Variables relevant to this method
+      // Return relevant string or JSON object
+    );
+    
+    // Main variant method - 'question' or 'content'
+    local question(variantParam) = {
+      // Variant JSON
+    };
+    
+    {
+      // Block JSON
+    }
+    ```
+- Question should be called with question(isProxy=true) or question(isProxy=false):
+```
+{
+  type: 'Question',
+  id: 'confirm_dob',
+  question_variants: [
+    {
+      question: question(isProxy=false)
+    },
+    {
+      question: question(isProxy=true)
+    },
+  ],
+}
+```
+- In the question method, when we need to resolve title, we call a new questionTitle method, passing isProxy:
```

Comment:
This is a good example, but before this we should explain what the standard is, then provide the example. 

Here we should say first that any time we want to vary the value of a property in a variant a method needs to be defined that returns the appropriate variant content.

Reply:
Paraphrased your text, thanks!👍